src/main/java/seedu/weme/logic/commands/memecommand/MemeViewCommand.java

```diff
@@ -0,0 +1,67 @@
+package seedu.weme.logic.commands.memecommand;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.weme.model.ModelContext.CONTEXT_VIEW;
+
+import java.util.List;
+
+import seedu.weme.commons.core.Messages;
+import seedu.weme.commons.core.index.Index;
+import seedu.weme.logic.commands.Command;
+import seedu.weme.logic.commands.CommandResult;
+import seedu.weme.logic.commands.exceptions.CommandException;
+import seedu.weme.model.Model;
+import seedu.weme.model.ModelContext;
+import seedu.weme.model.meme.Meme;
+
+/**
+ * Views a meme identified using it's displayed index from Weme.
+ */
+public class MemeViewCommand extends Command {
+
+    public static final String COMMAND_WORD = "view";
+
+    public static final String MESSAGE_DESCRIPTION = COMMAND_WORD
+            + ": views the meme identified by the index number used in the displayed meme list.";
+
+    public static final String MESSAGE_USAGE = MESSAGE_DESCRIPTION
+            + "\nParameters: INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " 1";
+
+    public static final String MESSAGE_VIEW_MEME_SUCCESS = "Viewing meme: %1$s";
+
+    private final Index targetIndex;
+
+    private ModelContext context = CONTEXT_VIEW;
+
+    public MemeViewCommand(Index targetIndex) {
+        this.targetIndex = targetIndex;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Meme> lastShownList = model.getFilteredMemeList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_MEME_DISPLAYED_INDEX);
+        }
+
+        Meme memeToView = lastShownList.get(targetIndex.getZeroBased());
+        model.setViewableMeme(memeToView);
+
+        CommandResult result = new CommandResult(String.format(MESSAGE_VIEW_MEME_SUCCESS, memeToView));
+        model.setContext(context);
```

Comment:
rather than putting context as a variable, it should be clearer if you just pass CONTEXT VIEW directly here